lib/nve/nve-main-page.dart

```diff
@@ -46,9 +46,25 @@ class FirstScreen extends State<MyStatefulWidget> {
     MyStatefulGreenPage()
   ];
 
+  Future<void> isOutOfBounds() async {
+    double distance = await geo.distanceBetween(
+        _locationData.latitude,
+        _locationData.longitude,
+        Provider.of<PositionService>(context, listen: false).latitude,
+        Provider.of<PositionService>(context, listen: false).longitude);
+    if (distance > settings.boundary) {
+      debugPrint("The user goes out of bounds");
+    }
+  }
+
   void _onItemTapped(int index) {
     setState(() {
       _selectedIndex = index;
+      if (settings.enableAlerts == true && settings.boundary != null && 
```

Comment:
avoid making processes inside an UI event

to track location changes use

location.onLocationChanged.listen((LocationData currentLocation) {
  // Use current location
});